src/azure-cli/azure/cli/command_modules/botservice/custom.py

```diff
@@ -75,6 +76,28 @@ def __prepare_configuration_file(cmd, resource_group_name, kudu_client, folder_p
             f.write(json.dumps(existing))
 
 
+def __handle_failed_name_check(name_response, cmd, client, resource_group_name, resource_name):
+    # Creates should be idempotent, verify if the bot already exists inside of the provided Resource Group
+    logger.debug('Failed name availability check for provided bot name "%s".\n'
+                 'Checking if bot exists in Resource Group "%s".', resource_name, resource_group_name)
+    try:
+        # If the bot exists, return the bot's information to the user
+        existing_bot = get_bot(cmd, client, resource_group_name, resource_name)
+        logger.warning('Provided bot name already exists in Resource Group. Returning bot information:')
+        return existing_bot
+    except ErrorException as e:
+        if e.error.error.code == 'ResourceNotFound':
+            code = e.error.error.code
+            message = e.error.error.message
+
+            logger.debug('Bot "%s" not found in Resource Group "%s".\n  Code: "%s"\n  Message: '
+                         '"%s"', resource_name, resource_group_name, code, message)
+            raise CLIError('Unable to create bot.\nReason: "{}"'.format(name_response.message))
```

Comment:
Just out of curiosity, if the bot doesn't exist, shouldn't `name_response.valid == True`? It may not reach here at all? Sorry if I miss anything.

Reply:
This is correct, we only enter this logic if the response from the name availability check indicates an unavailable name. This internal method would not be called if `name_response.valid == True`

[diffs L116-L117](https://github.com/Azure/azure-cli/pull/11935/files/c3955fdc430fe78e2864efec780bcbd567bb2047#diff-29e4a6bde35880c6917ba0382e947063R116-R117)